### PR TITLE
feat(projection-converter) - made the projection converter into a service

### PR DESCRIPTION
**Pull request by Danaorlov1005**
PLEASE APPROVE 🙍

### CODE REVIEW

File: confd/production.tmpl [hunk 1]
```diff
@@ -136,7 +136,11 @@
                 "datum": "ed50",
                 "projection": "utm"
             }
-        }
+        },
+		"Proj4":{
```
Comment (AmitMY) [inline confd/production.tmpl]: space

File: confd/production.tmpl [hunk 2]
```diff
@@ -136,7 +136,11 @@
...
+			"ed50":"+proj=utm +datum=ed50 +zone=${zone} +ellps=intl +units=m + no_defs",
```
Comment (AmitMY) [inline confd/production.tmpl]: space

File: confd/production.tmpl [hunk 3]
```diff
@@ -136,7 +136,11 @@
...
+			"ed5036" : ""
```
Comment (AmitMY) [inline confd/production.tmpl]: remove space

File: src/app/packages/core/services/projection-converter.service.ts [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+		const toWgs84Geo = to.datum === 'wgs84' && to.projection === 'geo';
+		const fromEd50Utm = from.datum === 'ed50' && from.projection === 'utm';
+		const toEd50Utm = to.datum === 'ed50' && to.projection === 'utm';
+
+
+		if (fromWgs84Geo && toEd50Utm) {
+			const lng = coords[0];
+			const hemisphere = coords[1];
+			const zoneUtmProj = this.getZoneUtmProj(lng, hemisphere);
+			console.log(zoneUtmProj.utmProj);
```
Comment (AmitMY) [inline src/app/packages/core/services/projection-converter.service.ts]: no

File: src/app/packages/core/services/projection-converter.service.ts [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+			const conv = (<any>proj4).default('EPSG:4326', zoneUtmProj.utmProj, coords);
+			return [...conv, zoneUtmProj.zone];
+		}
+
+		if (fromEd50Utm && toWgs84Geo) {
+			const zone = coords[2];
+			const utmProj = this.toolsConfig.Proj4.ed50.replace('${zone}', zone.toString());
+			console.log(utmProj + 'dfdfd');
```
Comment (AmitMY) [inline src/app/packages/core/services/projection-converter.service.ts]: no

File: src/assets/config/app.config.json [hunk 6]
```diff
@@ -163,6 +163,10 @@
 				"datum": "ed50",
 				"projection": "utm"
 			}
+		},
+		"Proj4":{
```
Comment (AmitMY) [inline src/assets/config/app.config.json]: space

File: src/assets/config/app.config.json [hunk 7]
```diff
@@ -163,6 +163,10 @@
...
+			"ed50":"+proj=utm +datum=ed50 +zone=${zone} +ellps=intl +units=m + no_defs",
```
Comment (AmitMY) [inline src/assets/config/app.config.json]: space

File: src/assets/config/app.config.json [hunk 8]
```diff
@@ -163,6 +163,10 @@
...
+			"ed5036" : ""
```
Comment (AmitMY) [inline src/assets/config/app.config.json]: remove space

**Comment by AmitMY** [comment]
And remove `covert-projections.ts`

**Comment by Danaorlov1005** [comment]
thx

File: src/assets/config/app.config.json (hunk 7 above)
Comment (Danaorlov1005) [reply src/assets/config/app.config.json]: ok

File: src/assets/config/app.config.json (hunk 8 above)
Comment (Danaorlov1005) [reply src/assets/config/app.config.json]: ok

File: src/assets/config/app.config.json (hunk 6 above)
Comment (Danaorlov1005) [reply src/assets/config/app.config.json]: ok

File: src/app/packages/core/services/projection-converter.service.ts (hunk 4 above)
Comment (Danaorlov1005) [reply src/app/packages/core/services/projection-converter.service.ts]: ok

File: confd/production.tmpl (hunk 3 above)
Comment (Danaorlov1005) [reply confd/production.tmpl]: ok

File: confd/production.tmpl (hunk 2 above)
Comment (Danaorlov1005) [reply confd/production.tmpl]: ok

File: confd/production.tmpl (hunk 1 above)
Comment (Danaorlov1005) [reply confd/production.tmpl]: ok

File: src/app/packages/menu-items/tools/go-to/go-to/go-to.component.spec.ts [hunk 9]
```diff
@@ -5,7 +5,7 @@ import { Store, StoreModule } from '@ngrx/store';
 import { IToolsState, toolsFeatureKey, ToolsReducer } from '../../reducers/tools.reducer';
 import { GoToModule } from '../go-to.module';
 import { GoToAction, SetPinLocationModeAction } from '../../actions/tools.actions';
-import * as utilCovertProjections from '@ansyn/core/utils/covert-projections';
+import { CoreModule } from '../../../../core';
```
Comment (AmitMY) [inline src/app/packages/menu-items/tools/go-to/go-to/go-to.component.spec.ts]: package

File: src/app/packages/menu-items/tools/go-to/go-to/go-to.component.spec.ts [hunk 10]
```diff
@@ -41,14 +41,14 @@ describe('GoToComponent', () => {
...
 	describe('submitGoTo', () => {
-		it('should dispatch GoToAction with convertByProjectionDatum result', () => {
-			spyOn(store, 'dispatch');
-
-			spyOn(utilCovertProjections, 'convertByProjectionDatum').and.returnValue([0, 0]);
-
-			component.submitGoTo();
-			expect(store.dispatch).toHaveBeenCalledWith(new GoToAction([0, 0]));
-		});
+		// it('should dispatch GoToAction with convertByProjectionDatum result', () => {
```
Comment (AmitMY) [inline src/app/packages/menu-items/tools/go-to/go-to/go-to.component.spec.ts]: uncomment

File: src/app/packages/core/services/projection-converter.service.spec.ts [hunk 11]
```diff
@@ -0,0 +1,48 @@
...
+
+	it('should be created', () => {
+		expect(projectionConverterService).toBeTruthy();
+	});
+
+	it('should convert wgs84geo', () => {
+		const result = projectionConverterService.convertByProjectionDatum([lonwgs84, latwgs84],
+			{ datum: 'wgs84', projection: 'geo' },
+			{ datum: 'ed50', projection: 'utm' }).map((num) => Math.floor(num));
+		expect(result).toEqual([loned50, latd50, 18]);
```
Comment (yuvalbl) [inline src/app/packages/core/services/projection-converter.service.spec.ts]: set as a variable

File: src/app/packages/core/services/projection-converter.service.spec.ts [hunk 12]
```diff
@@ -0,0 +1,48 @@
...
+	});
+
+	it('should convert ed50', () => {
+		const result = projectionConverterService.convertByProjectionDatum([loned50, latd50, 18],
```
Comment (yuvalbl) [inline src/app/packages/core/services/projection-converter.service.spec.ts]: set 18 as a constant

File: src/app/packages/core/services/projection-converter.service.spec.ts [hunk 13]
```diff
@@ -0,0 +1,48 @@
+import { inject, TestBed } from '@angular/core/testing';
+
+import { ProjectionConverterService } from './projection-converter.service';
+import { toolsConfig } from '@ansyn/menu-items/tools/models';
+
+describe('ProjectionConverterService', () => {
+	let projectionConverterService: ProjectionConverterService;
```
Comment (yuvalbl) [inline src/app/packages/core/services/projection-converter.service.spec.ts]: let  => const

File: src/app/packages/core/services/projection-converter.service.ts [hunk 14]
```diff
@@ -0,0 +1,51 @@
+import { isEqual as _isEqual } from 'lodash';
+import * as proj4 from 'proj4';
```
Comment (yuvalbl) [inline src/app/packages/core/services/projection-converter.service.ts]: import explicitly

File: src/app/packages/core/services/projection-converter.service.ts [hunk 15]
```diff
@@ -0,0 +1,51 @@
...
+
+		if (fromEd50Utm && toWgs84Geo) {
+			const zone = coords[2];
+			const utmProj = this.getUtmFromConf(zone);
+			const conv = (<any>proj4).default(utmProj, 'EPSG:4326', [coords[0], coords[1]]);
+			return [...conv];
+		}
+	}
+
+	getZoneUtmProj(lng: number, hemisphere: number): UtmZone {
```
Comment (yuvalbl) [inline src/app/packages/core/services/projection-converter.service.ts]: add the following comment:
// leave hemisphere alone for future use :)

File: src/app/packages/core/services/projection-converter.service.ts [hunk 16]
```diff
@@ -0,0 +1,51 @@
...
+		const zone = (Math.floor((lng + 180) / 6) % 60) + 1;
```
Comment (yuvalbl) [inline src/app/packages/core/services/projection-converter.service.ts]: add link to source OR
change "magic numbers" to const

File: src/app/packages/overlays/package.json [hunk 17]
```diff
@@ -9,6 +9,6 @@
 	"author": "",
 	"license": "MIT",
 	"peerDependencies": {
-		"ansyn-event-drops": "^0.4.12"
```
Comment (yuvalbl) [inline src/app/packages/overlays/package.json]: do rebase

**Pull request merged**